infra/general/kEmails.php

```diff
@@ -75,9 +76,16 @@ protected static function getFormattedEmailComponentName($blockType, $roleName)
 		return $blockType . '-' . $roleName;
 	}
 	
-	public static function getDynamicTemplateBaseLink($roleName)
+	public static function getDynamicTemplateBaseLink($roleName, $is2FA = false)

```

Comment:
we should implement the function differently as we will probably have it for sso as well
is2FA is too specific, we should have it with $baseLink = self::DYNAMIC_EMAIL_2FA_BASE_LINK by default and keep the old logic, you can pass other type from outside that will override